### ISSUE TITLE
Training Bonito on Heavily Modified DNA samples

### ISSUE THREAD

**Issue by jorisbalc** [nanoporetech/bonito#367]
Hello,

I've attempted to train/finetune a bonito model on a specific sample I'd like to use for the preparation of a remora model. Since remora workflow now requires the appropriate calls within a .bam file, I was encouraged to finetune a bonito model to gather more accurate basecalls. Currently, the super high accuracy model (via newest dorado version, and kit14 chemistry) gives me at most 82% accuracy on the modified DNA, which results in a usable, yet poor quality remora model (since some of the modified Cs get mismatched). I trained a bonito model from scratch using my data, but it seems like that doesn't do much. The accuracy is pretty much the same, albeit the errors are less likely to occur near CG sites and are rather distributed all throughout the reads. The finetuning option gave a different outcome and most of the CG sites simply have deletions in them resulting in a read length loss of over 50% in just one epoch of training. I'm attaching some IGV pictures to visualize the situation more vividly. 

The mismatches are by the synthetic modification, the samples themselves are relatively clean and we had no problem preparing Remora models within GCGC context (models were of 95+ accuracy, albeit the amount of mismatches near GCGC sites were the same as in CG sites). With these inaccurate basecalls, CG context Remora model cannot provide an accurate model due to the mismatches I assume. The rate of false positives are below 0.02, while the rate of true positives does not go above .6. 

Any tips or ideas are appreciated!

<details>
<summary>Canonical versus modified calls</summary>

![image](https://github.com/nanoporetech/bonito/assets/110527583/abec964d-c49e-48d5-b42a-2c98c15e7d6e)

</details>

<details>
<summary>Modified calls and modified calls after finetuning</summary>

![image](https://github.com/nanoporetech/bonito/assets/110527583/1672e336-8761-4fb1-a865-01cdcedec545)

</details>

EDIT: solved

**Comment by jorisbalc** [comment]
Hey @touala,

For the training, I used ctc data with called with similar accuracy as regular basecalls. The default in bonito save-ctc-data is .99 if I recall correctly. I assume the reason behind the model trained in deletions is the high accuracy calls in the ctc data contain quite more deletions than mismatches. The lower accuracy ctc data (I used min-save-ctc 0.80) resulted me in a fairly decent model which managed to basecall the modified bases correctly.

**Comment by touala** [comment]
Hi @jorisbalc,

Thank you for taking the time to respond. Any other tricks downstream? I've experiment with `--min-accuracy-save-ctc` and `bonito train --epochs 1 --lr 5e-4 --pretrained dna_r10.4.1_e8.2_400bps_sup@v4.3.0` but I'm not convinced that this is optimal.  

Alan
